File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/statistik/Application.java
package de.maibornwolff.domainchoreograph.examples.statistik;

import de.maibornwolff.domainchoreograph.examples.statistik.domaintypes.GesamtAuswertung;
import de.maibornwolff.domainchoreograph.examples.statistik.domaintypes.Statistik;
import de.maibornwolff.domainchoreograph.core.api.DomainEnvironment;
import de.maibornwolff.domainchoreograph.examples.statistik.choreographies.GesamtAuswertungCalculator;

public class Application {

    public static void main(String[] args) {
        DomainEnvironment environment = DomainEnvironment.builder()
            .addLogger(ChoreographyDevtool.domainAnalytics.getLogger())
            .build();
        GesamtAuswertungCalculator calculator = environment.get(GesamtAuswertungCalculator.class);
        GesamtAuswertung gesamtAuswertung = calculator.calculate(new StichprobenQuelle());

        for (Statistik statistik : gesamtAuswertung.getStatistikList()) {
            System.out.println(statistik);
            System.out.println("----");
        }

        ChoreographyDevtool.domainAnalytics.getServer().start();
    }
}

File: core/src/test/java/de/maibornwolff/domainchoreograph/example/Application.kt
package de.maibornwolff.domainchoreograph.example

import de.maibornwolff.domainchoreograph.core.api.DomainEnvironment
import de.maibornwolff.domainchoreograph.example.choreographies.GesamtAuswertungChoreography

object Application {

    @JvmStatic
    fun main(args: Array<String>) {

        val gesamtAuswertungChoreography = DomainEnvironment().get(
            GesamtAuswertungChoreography::class.java
        )

        val (statistikList) = gesamtAuswertungChoreography.calculate(StichprobenQuelle())

        for (statistik in statistikList) {
            println(statistik)
            println("----")
        }
    }
}
